SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -1,19 +1,70 @@
 package com.jetbrains.handson.mpp.mobile
 
+import io.ktor.client.HttpClient
+import io.ktor.client.features.json.JsonFeature
+import io.ktor.client.features.json.serializer.KotlinxSerializer
+import io.ktor.client.request.get
+import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.readText
 import kotlinx.coroutines.*
+import kotlinx.serialization.ImplicitReflectionSerializer
+import kotlinx.serialization.UnstableDefault
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import kotlinx.serialization.json.JsonElement
 import kotlin.coroutines.CoroutineContext
 
 class ApplicationPresenter: ApplicationContract.Presenter() {
 
     private val dispatchers = AppDispatchersImpl()
     private var view: ApplicationContract.View? = null
     private val job: Job = SupervisorJob()
-
+    private val codeMap = mapOf<String, String>("Harrow and Wealdstone" to  "HRW",
+        "Canley" to "CNL",
+        "London Euston" to "EUS",
+        "Coventry" to "COV",
+        "Birmingham New Street" to "BHM")
     override val coroutineContext: CoroutineContext
         get() = dispatchers.main + job
 
     override fun onViewTaken(view: ApplicationContract.View) {
         this.view = view
-        view.setLabel(createApplicationScreenMessage())
+        view.setLabel("Get live train times")
+    }
+
+    @ImplicitReflectionSerializer
+    @OptIn(UnstableDefault::class)
+
+    override fun onButtonPressed(origin: String, destination: String) {
+        val originCode = requireNotNull(codeMap[origin])
+        val destinationCode = requireNotNull(codeMap[destination])
```

Comment:
Probably worth doing some validation (say, make button disabled if any of codes are null) - but can be done in a separate PR, because it will be a bit of work.